corehq/apps/toggle_ui/views.py

```diff
@@ -94,10 +111,14 @@ def post(self, request, *args, **kwargs):
             item_list = json.loads(item_list)
             item_list = [u.strip() for u in item_list if u and u.strip()]
 
-        affected_users = set(toggle.enabled_users) | set(item_list)
+        old_entries = set(toggle.enabled_users)
+        new_entries = set(item_list)
         toggle.enabled_users = item_list
         toggle.save()
-        for item in affected_users:
+
+        changed_entries = old_entries ^ new_entries  # ^ is XOR in python
+        self.call_save_fn(changed_entries, new_entries)
```

Comment:
Effectively since call_save_fn is acting only on newly added entries. You could pass old_entries, new_entries and use `new - old` to apply master-enable and `old-new` to reverse the master-enable (which I think `save_fn` should be able to decide, depending on the toggle)